controllers/machine_controller.go

```diff
@@ -75,10 +78,18 @@ func (r *MachineReconciler) SetupWithManager(mgr ctrl.Manager, options controlle
 		WithOptions(options).
 		Build(r)
 
+	if err != nil {
+		return errors.Wrap(err, "failed setting up with a controller manager")
+	}
+
 	r.controller = c
 	r.recorder = mgr.GetEventRecorderFor("machine-controller")
 	r.config = mgr.GetConfig()
-	return err
+
+	r.scheme = runtime.NewScheme()
```

Comment:
What's the reasoning behind using a new scheme instead of using scheme.Scheme, which we already initialize with our types in `main.go` and the test suites?

Reply:
I had suggested to Noa that we set up a scheme that each reconciler could store for use specifically with the remote cluster client, and that each scheme would only have the necessary types added to it based on how the remote cluster client is used per reconciler.

Reply:
Is there any downside in reusing the same one?

Reply:
We generally have a single `runtime.Scheme`, defined in `main.go`, but it's a package-level variable, and it's not exported (nor do I think it should be, at least not as a package-level variable).

However, it is available via `mgr.GetScheme()` in all the `SetupWithManager()` functions, so we could probably just store a reference to it in each reconciler. Would that work for you?

Reply:
I was mostly trying to understand the reasoning for posterity and best practices. I was under the impression that re-using one scheme throughout the lifecycle of a controller was the way to go. I'm ok either way, but we should make sure to update the other controllers as well if we go down this approach

Reply:
I think Noa did update all the other reconcilers that use `remote.NewClusterClient`, right?

I like the idea of using the manager's scheme whenever our reconcilers create a new remote cluster client.